website/common/script/libs/getOfficialPinnedItems.js

```diff
@@ -1,10 +1,26 @@
-import toArray from 'lodash/toArray';
 import content from '../content/index';
 import SeasonalShopConfig from './shops-seasonal.config';
 
 const { officialPinnedItems } = content;
 
-const flatGearArray = toArray(content.gear.flat);
+function groupBySet (items) {
+  const grouped = {};
+
+  Object.keys(items).forEach(key => {
+    const item = items[key];
+    const { set } = item;
+
+    if (!grouped[set]) {
+      grouped[set] = [];
+    }
+
+    grouped[set].push(item);
+  });
+
+  return grouped;
+}
+
+const flatGearArray = groupBySet(content.gear.flat);
```

Comment:
could you try it as well with lodash's groupBy? since we have lodash already in many multiple places then we wouldn't need to write a custom groupBy method :)

Reply:
I wonder if we have more places that needs this access by set, but for now this a very nice start 👍

Reply:
usually lodash is slow, i will do a benchmark and implement the fastest

Reply:
Lodash x 2,687 ops/sec ±1.49%
groupBySet x 2,534 ops/sec ±4.15%

``` js
import _ from 'lodash';
import { Suite } from 'benchmark';
[....]
// eslint-disable-next-line new-parens
const suite = new Suite;
suite
  .add('Lodash', () => {
    _.groupBy(content.gear.flat, 'set');
  })
  .add('groupBySet', () => {
    groupBySet(content.gear.flat);
  });

suite
  .on('cycle', event => {
    console.log(String(event.target));
  })
  .on('complete', () => {
    console.log('Benchmark completo');
  })
  .run({ async: true });

[...]
```

the self-implemented code is prettry faster
what i can do is  make it generic and move it for some "util" folder

Reply:
Lodash x 2,687 ops/sec ±1.49%
groupBySet x 2,534 ops/sec ±4.15%

lodash wins